src/mongoc/mongoc-write-command.c

```diff
@@ -780,13 +809,15 @@ _mongoc_write_command_execute (mongoc_write_command_t       *command,       /* I
 
    command->hint = hint;
 
-   node = &client->cluster.nodes [hint - 1];
-   mode = SUPPORTS_WRITE_COMMANDS (node);
+   server = mongoc_topology_server_by_id(client->topology, hint);
+   mode = (server->min_wire_version <= 2 && server->max_wire_version >= 2);
```

Comment:
Thats a very magical 2. I liked the macro better :/
Maybe define 2 as WRITE_COMMANDS or something?